### ISSUE TITLE
FPE build causes error in colors unit test

### ISSUE THREAD

**Issue by warrickball** [MESAHub/mesa#395]
When I tried to do some local FPE testing with `MESA_FPE_CHECKS_ON=1`, I hit this failure in the `colors` unit tests:
```
At line 1191 of file ../public/utils_lib.f90
Fortran runtime error: Substring out of bounds: upper bound (4097) of 'line' exceeds string length (4096)

Error termination. Backtrace:
#0  0x41111a in __utils_lib_MOD_split_line
        at ../public/utils_lib.f90:1191
#1  0x40b67c in read_one_colors_data
        at ../private/mod_colors.f90:248
#2  0x40a9f2 in read_colors_data
        at ../private/mod_colors.f90:328
#3  0x40ca90 in init_colors
        at ../private/mod_colors.f90:102
#4  0x40d78b in __mod_colors_MOD_do_colors_init
        at ../private/mod_colors.f90:84
#5  0x407478 in __colors_lib_MOD_colors_init
        at ../public/colors_lib.f90:74
#6  0x403e3a in do_test_colors
        at ../src/test_colors.f90:33
#7  0x40261b in test_colors
        at ../src/test_colors.f90:10
#8  0x403f20 in main
        at ../src/test_colors.f90:3
```
The offending line is in the new `split_line` subroutine:
https://github.com/MESAHub/mesa/blob/ac0bbdd17fe5b05439c737300f30eb250841168d/utils/public/utils_lib.f90#L1175-L1207
and I think I see the problem. IIRC Fortran doesn't specify the order in which logicals are evaluated, so in the expression
```
line(k:k)==' ' .and. line(k+1:k+1)==' ' .and. k < len(line)
```
 we might try to evaluate `line(k+1:k+1)` before we've actually checked that `k < len(line)`. (Changing the order in the source changes nothing.) So it makes sense that we can hit the index-out-of-bounds error, because `MESA_FPE_CHECKS_ON=1` adds `-fbounds-check` to the compiler flags.

I was able to fix the code by nesting the logical blocks inside `k < len(line)`, i.e.,
```f90
            if(k < len(line)) then
               if(line(k:k)==' ' .and. line(k+1:k+1)==' ') then
                  k = k+1
                  cycle inner
               end if
            end if
```
I'm not sure if there's a more elegant/sensible solution and anyway I'm worried about the next block inside the inner loop, which has the logical expression:
```
line(k:k)==' ' .or. k > len(line)
```
which also seems like it's waiting for another index-out-of-bounds if we try `line(k:k)` before we've checked `k` (though perhaps we can never reach that case).

I would open a PR but my guess is @rjfarmer has a clearer idea of what should happen here.

**Comment by rjfarmer** [comment]
Yes that's the solution. I'd already fixed it in my rf/rates2 branch ff6c2bb I just hadn't merged it yet.

**Comment by rjfarmer** [comment]
Go ahead and cherry-pick
